smarts/core/agent_manager.py

```diff
@@ -213,6 +213,10 @@ def observe(self, sim):
                 rewards[agent_id] = vehicle.trip_meter_sensor(increment=True)
                 scores[agent_id] = vehicle.trip_meter_sensor()
 
+        if sim.should_reset:
+            dones = {agent_id: True for agent_id in self.agent_ids}
+            dones["__sim__"] = True
```

Comment:
The idea here is that we could treat the sim as also having the possibility to be done.

Reply:
It can come through the `dones` from `SMARTS`; however, at the gym level it would need to be translated.